do/serverless.go

```diff
@@ -131,21 +131,44 @@ type Annotation struct {
 	Value interface{} `json:"value"`
 }
 
-// ServerlessProject ...
+// ServerlessProject provides a "physical" view of a project (its files and directories).
+// The "logical" view is provided by the ProjectSpec, which is formed by parsing the file
+// at ConfigPath (if any), and by interpreting the structure in ServerlessProject.
+// This structure is never serialized.
 type ServerlessProject struct {
-	ProjectPath string   `json:"project_path"`
-	ConfigPath  string   `json:"config"`
-	Packages    string   `json:"packages"`
-	Env         string   `json:"env"`
-	Strays      []string `json:"strays"`
+	ProjectPath string                 `json:"project_path"`
+	ConfigPath  string                 `json:"config"`
+	Packages    []ServerlessPackageDir `json:"packages"`
+	EnvPath     string                 `json:"env_path"`
+	LibPath     string                 `json:"lib_path"`
 }
 
-// ProjectSpec describes a project.yml spec
+// ServerlessPackageDir provides the contents of one package under ServerlessProject
+type ServerlessPackageDir struct {
+	Path      string
+	Functions []ServerlessFunctionSource
+}
+
+// FunctionSource provides the contents of one function in a PackageDir.
+// Because a function can be represented by either a directory or a (single-source) file,
+// we represent each function as a set of paths.   This may be one path (e.g. Name=foo,
+// Paths=[/path/to/foo.js].   Or multiples (e.g. Name=foo, Paths=[/path/to/foo/index.js,
+// /path/to/foo/package.json, /path/to/foo/.include, ...]).
+type ServerlessFunctionSource struct {
+	Name  string
+	Paths []string
+}
+
+// ProjectSpec describes a project.yml spec (both initially and as subsequently augmented by
+// structural information about the project).
 // reference: https://docs.nimbella.com/configuration/
+// TODO the FileStructure member is temporary, since the work to merge file/directory structure in
+// ServerlessProject with the spec information here is still pending.
 type ProjectSpec struct {
-	Parameters  map[string]interface{} `json:"parameters,omitempty"`
-	Environment map[string]interface{} `json:"environment,omitempty"`
-	Packages    []*ProjectSpecPackage  `json:"packages,omitempty"`
+	Parameters    map[string]interface{} `json:"parameters,omitempty"`
+	Environment   map[string]interface{} `json:"environment,omitempty"`
+	Packages      []*ProjectSpecPackage  `json:"packages,omitempty"`
+	FileStructure *ServerlessProject     `json:"file_sructure,omitempty"`
```

Comment:
As the comment above states, the `FileStructure` member is temporary pending the implementation of a proper merge of information from the file structure with information from the `project.yml` file.    What's in the `FileStructure` information needs to be absorbed into the `Packages` member of the `ProjectSpec`